File: back/src/main/java/com/example/ss1/services/MailService.java
package com.example.ss1.services;

import java.security.SecureRandom;

import org.springframework.core.io.FileSystemResource;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;
import org.springframework.stereotype.Service;

import com.example.ss1.models.Usuario;

import jakarta.mail.internet.MimeMessage;

@Service
public class MailService {
    private final JavaMailSender mailSender;
    private static final SecureRandom random = new SecureRandom();
    public MailService(JavaMailSender mailSender){
        this.mailSender = mailSender;
    }

      // Texto plano
    public void sendText(String to, String subject, String text) {
        SimpleMailMessage msg = new SimpleMailMessage();
        msg.setTo(to);
        msg.setSubject(subject);
        msg.setText(text);
        mailSender.send(msg);
    }

    // HTML + adjuntos
    public void sendHtml(String to, String subject, String html, String pathAdjunto) throws Exception {
        MimeMessage mime = mailSender.createMimeMessage();
        MimeMessageHelper helper = new MimeMessageHelper(mime, true, "UTF-8");
        helper.setTo(to);
        helper.setSubject(subject);
        helper.setText(html, true); // true = HTML

        if (pathAdjunto != null) {
            FileSystemResource file = new FileSystemResource(pathAdjunto);
            helper.addAttachment(file.getFilename(), file);
        }
        mailSender.send(mime);
    }

    public void enviarCodigo(Usuario usuario, String codigo) {
        String cuerpo = """
                Hola,

                Este es tu código de verificación de correo electrónico:

                %s

                Si tú no creaste esta cuenta, puedes ignorar este correo.
                """.formatted(codigo);

        sendText(
                usuario.getEmail(),
                "Confirmación de correo electrónico",
                cuerpo);
    }

    public String generarCodigo() {
        int numero = random.nextInt(1_000_000); // 0 a 999999
        return String.format("%06d", numero);
    }
}

File: back2/api/v1/services/mail_service.py
import os
import smtplib
from email.message import EmailMessage
from secrets import randbelow

class MailService:
    def __init__(self):
        self.mail_user = os.getenv("MAIL_USER")
        self.mail_pass = os.getenv("MAIL_PASS")  # app password (sin espacios)
        if not self.mail_user or not self.mail_pass:
            raise RuntimeError("Faltan variables de entorno MAIL_USER/MAIL_PASS")

    def generar_codigo(self) -> str:
        return f"{randbelow(1_000_000):06d}"

    def send_text(self, to: str, subject: str, text: str):
        msg = EmailMessage()
        msg["From"] = self.mail_user
        msg["To"] = to
        msg["Subject"] = subject
        msg.set_content(text)

        with smtplib.SMTP("smtp.gmail.com", 587) as smtp:
            smtp.ehlo()
            smtp.starttls()
            smtp.login(self.mail_user, self.mail_pass)
            smtp.send_message(msg)

    def enviar_codigo(self, email: str, codigo: str):
        cuerpo = (
            "Hola,\n\n"
            "Este es tu código de verificación de correo electrónico:\n\n"
            f"{codigo}\n\n"
            "Si tú no creaste esta cuenta, puedes ignorar este correo.\n"
        )
        self.send_text(email, "Confirmación de correo electrónico", cuerpo)
